src/painter.rs

```diff
@@ -35,6 +35,54 @@ impl PromptCoordinates {
     }
 }
 
+pub struct PromptLines<'prompt> {
+    pub before_cursor_lines: Vec<&'prompt str>,
+    pub after_cursor_lines: Vec<&'prompt str>,
+    pub hint_lines: Vec<&'prompt str>,
+}
+
+impl<'prompt> PromptLines<'prompt> {
+    /// Splits the strings before and after the cursor as well as the hint
+    /// This vector with the str are used to calculate how many lines are
+    /// required to print after the prompt
+    pub fn from_strings(
+        before_cursor: &'prompt str,
+        after_cursor: &'prompt str,
+        hint: &'prompt str,
+    ) -> Self {
+        let before_cursor_lines = if cfg!(windows) {
+            before_cursor.split("\r\n").collect::<Vec<&str>>()
+        } else {
+            #[allow(clippy::single_char_pattern)]
+            before_cursor.split("\n").collect::<Vec<&str>>()
+        };
+
+        let after_cursor_lines = if cfg!(windows) {
+            after_cursor.split("\r\n").collect::<Vec<&str>>()
+        } else {
+            #[allow(clippy::single_char_pattern)]
+            after_cursor.split("\n").collect::<Vec<&str>>()
+        };
+
+        let hint_lines = if cfg!(windows) {
+            hint.split("\r\n").collect::<Vec<&str>>()
+        } else {
+            #[allow(clippy::single_char_pattern)]
+            hint.split("\n").collect::<Vec<&str>>()
+        };
```

Comment:
For better readability how about making the switch in the definition of a single `const NEW_LINE_SEPARATOR: &str` ?

Reply:
Alternatively... is `str::lines` doing the right thing already?